src/textual/app.py

```diff
@@ -1978,8 +1978,8 @@ async def invoke_ready_callback() -> None:
                                 with redirect_stdout(redirector):  # type: ignore
                                     await run_process_messages()
                         else:
-                            with redirect_stderr(None):
-                                with redirect_stdout(None):
+                            with redirect_stderr(open(os.devnull, 'w')):
```

Comment:
The file handle should really be explicitly closed. You could probably get away with opening a single file.

Reply:
You're absolutely right, sorry about that.

Reply:
No worries!